### PR TITLE
documentation deepallDIAL

### PR DESCRIPTION
**Pull request by Car-la-F**
will solve #65

### CODE REVIEW

**Comment by Car-la-F** [comment]
@smilesun 
I have the fealing there are more differences between the implementation described in the paper and the implementations in domainlab. For instance the network structure (as it is described in the paper) cannot be fully trained with using the loss function 

$$
 \text{loss} = \text{loss on original images} + \gamma \text{ loss on adversary images}
$$

because in the paper they do use a external domain classifier (to classify natural image an adversary image) which is not respected in this loss function.

The network structure proposed in the paper is also documented in the markdown file right now. Can you check if this is the network implemented in Domainlab and if not, what is implemented instead?

**Comment by Car-la-F** [comment]
From what I understand from running the `test_trainer_dial.py`. The network is initialised in 

https://github.com/marrlab/DomainLab/blob/5b039812f427c707a431a586cc1140e0e5b137d9/domainlab/algos/builder_dial.py#L38

running the pytest function it corresponds to the "conv_bn_pool_2" network, which is also expected because this is specified in the args. 

This would mean that there is actually no domain classifier involved in the whole training.

**Comment by smilesun** [comment]
thanks for clarifying,  we could combine dial with other models in domainlab (currently this is available) so missing that part is not that important, we do not have to stick to that.

**Comment by smilesun** [comment]
> @smilesun
> 
> I started working on the documentation for DIAL. Doing so I got some questions:
> 
> 1. the paper proposes two loss functions, one using only the classical cross entropy loss and one using the Kullback-Leibler divergence. Can one control which loss is used in Domainlab?
> 2. The paper uses two hyperparameter (r and λ) to weight the terms in the loss function. r weights the domain classification loss and λ the label classification loss in the adversary domains and the Kullback-Leibler divergence between the label output of natural domain and their adversary domains. In the code I can only find `gamma_reg` and I don't understand which term is seen as the regularisation term
> 
> https://github.com/marrlab/DomainLab/blob/0e64a4a2cf8bf30ec8d92de7b43964adf55f4f42/domainlab/algos/trainers/train_dial.py#L57
> 
> 3. shall I write something about the strategy of generating the images of the adversary domain?

currently we do not support the domain classification to make it simple.

**Comment by smilesun** [comment]
> From what I understand from running the `test_trainer_dial.py`. The network is initialised in
> 
> https://github.com/marrlab/DomainLab/blob/5b039812f427c707a431a586cc1140e0e5b137d9/domainlab/algos/builder_dial.py#L38
> 
> running the pytest function it corresponds to the "conv_bn_pool_2" network, which is also expected because this is specified in the args.
> 
> This would mean that there is actually no domain classifier involved in the whole training.

correct observation, we can have another issue refered to this, so we can close this one.
